iot-hub/Samples/device/FileUploadSample/FileUploadSample.cs

```diff
@@ -1,112 +1,66 @@
-
-                fileUploadTime.Stop();
-                return;
-            }
-
-            Console.WriteLine("Successfully uploaded the file to Azure Storage");
-
-            var successfulFileUploadCompletionNotification = new FileUploadCompletionNotification
-            {
-                // Mandatory. Must be the same value as the correlation id returned in the sas uri response
-                CorrelationId = sasUri.CorrelationId,
-
-                // Mandatory. Will be present when service client receives this file upload notification
-                IsSuccess = true,
-
-                // Optional, user defined status code. Will be present when service client receives this file upload notification
-                StatusCode = 200,
-
-                // Optional, user-defined status description. Will be present when service client receives this file upload notification
-                StatusDescription = "Success"
-            };
-
-            await _deviceClient.CompleteFileUploadAsync(successfulFileUploadCompletionNotification);
-            Console.WriteLine("Notified IoT Hub that the file upload succeeded and that the SAS URI can be freed.");
-
-            fileUploadTime.Stop();
-
-            Console.WriteLine($"Time to upload file: {fileUploadTime.Elapsed}.");
-        }
-    }
-}
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client.Transport;
+using Microsoft.WindowsAzure.Storage.Blob;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    public class FileUploadSample
+    {
+        private readonly DeviceClient _deviceClient;
+
+        public FileUploadSample(DeviceClient deviceClient)
+        {
+            _deviceClient = deviceClient;
+        }
+
+        public async Task RunSampleAsync()
+        {
+            const string filePath = "TestPayload.txt";
+
+            using (var fileStreamSource = new FileStream(FilePath, FileMode.Open))
+            {
+                var fileName = Path.GetFileName(fileStreamSource.Name);
+
+                Console.WriteLine("Uploading File: {0}", fileName);
+
+                var watch = Stopwatch.StartNew();
+
+                var sasUriReq = new Transport.FileUploadSasUriRequest() { BlobName = fileName };
+                var sasUriRes = await _deviceClient.GetFileUploadSasUriAsync(sasUriReq);
+                var isSuccess = false;
+
+                try {
+                    // Upload the file using BlobClient via HTTPS protocol, regardless of the DeviceClient protocol selection.
+                    var blobClient = new BlobClient(sasUriRes.GetBlobUri());
+
+                    var uploadRes = await blobClient.UploadAsync(fileStreamSource);
```

Comment:
Our latest sample already has this change.

https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/FileUploadSample/FileUploadSample.cs#L52